q.js

```diff
@@ -1502,6 +1506,10 @@ Promise.prototype.keys = function () {
 Q.all = all;
 function all(promises) {
     return when(promises, function (promises) {
+        if (!isArray(arguments[0]) || arguments.length !== 1) {
```

Comment:
I don't think it's good to prohibit extra arguments. Consider e.g. `arrayOfArrays.map(Q.all)`---that will now fail, since `map` passes `(array, index, arrayOfArrays)`.